### DISCUSSION TITLE
awesome suggestions

### DISCUSSION THREAD

**Discussion by LexterS999** [TechnitiumSoftware/DnsServer#277] [Ideas]
1) Make blocked ads more clear on logs section or create standalone tab which will show blocked requests. Right now, it's hard to see what was allowed/cached and blocked.

2) Provide more detailed instructions for cashing section. I.e. for moderate/aggressive cache settings

3) Let's add wildcard and regex type filters to block ads as custom and as URL filter

4) Let's make searching on cache/block/allow section zone more advanced. Right now I need to type whole domain including top level TLD's. Better if it will search only by keyword

5) Will be better if you can create more friendly cache/block/allow structure like in Pi-hole

6) And what about content filtering options?

Also, I'm interest if your DNS server solution allow bypassing CNAME to block ads?

**Comment by ShreyasZare** [comment]
Thanks for the suggestions.

>     1. Make blocked ads more clear on logs section or create standalone tab which will show blocked requests. Right now, it's hard to see what was allowed/cached and blocked.

There is a planned feature that will be added as an DNS App in a few updates. This app will log the queries into a database and this will be much better to filter and show results on the panel. 
 
>     2. Provide more detailed instructions for cashing section. I.e. for moderate/aggressive cache settings

Please elaborate this point to help me understand it better.

>     3. Let's add wildcard and regex type filters to block ads as custom and as URL filter

There is basic wildcard label support already. Regex support is planned. URL filter is not possible to do with DNS though.

>     4. Let's make searching on cache/block/allow section zone more advanced. Right now I need to type whole domain including top level TLD's. Better if it will search only by keyword

The data structure is internally a tree which make it browseable and query-able. Adding search would mean to scan entire tree which will be not be usable due to performance issues. So, it will need additional indexing which then requires implementing database like features. This is thus not feasible to implement.

>     5. Will be better if you can create more friendly cache/block/allow structure like in Pi-hole

I haven't used Pi-hole much. Will look into this to see how it can be improved.

>     6. And what about content filtering options?

Since this is a DNS server, only domain related filtering is possible with this project.

> Also, I'm interest if your DNS server solution allow bypassing CNAME to block ads?

If you are referring to CNAME cloaking, then yes the DNS server has this feature implemented. Which means any CNAME that points to a blocked domain name gets blocked too.

**Comment by LexterS999** [comment]
1) How it will be looked like that database?

2) I'm talking about to create blog post with cache settings for real life and normal/aggressive cache settings as for example

3) How to use wildcard, and what format and where to paste it?

> If you are referring to CNAME cloaking,

How to use it?

**Reply by ShreyasZare** [reply]
> 
>     1. How it will be looked like that database?

The DNS app will use either a local sqlite or user specified db server. There will be a tab in the web panel where there will be options to filter the records as per multiple criteria. So, it will be possible to select multiple filters and list records.
 
>     2. I'm talking about to create blog post with cache settings for real life and normal/aggressive cache settings as for example

There are no different modes for cache operation. The cache just keep records in memory till its TTL expires. If the serve stale feature is enabled, the expired records are not removed immediately and are used when the DNS server is unable to connect to the name server hosting that domain name. When serve stale TTL too expires then the record is removed from cache.

>     3. How to use wildcard, and what format and where to paste it?

Lets say you want to block all google domains like google.com or google.co.in then you can add "google.*" and "google.*.*" in the blocked zone and all matching domain names will get blocked.

> > If you are referring to CNAME cloaking,
> 
> How to use it?

CNAME cloaking is automatic. You don't have to do any config to make it work.

**Reply by LexterS999** [reply]
And what if I want to block, for example, subdomain.domain.com? Then what I need to write and where to paste it?

And what parameters do you recommend for more aggressive cashing? Also, I think it's better to keep stale records and other cache records on HDD/SDD instead of RAM. Because RAM is much more limited.

**Comment by LexterS999** [comment]
Also, can you tell me how to make ICANN root DNS to work? Do I need to choose none in quick select option, or just left it empty everywhere? And how much root DNS servers it's using, where I can find it on my VPS?

**Reply by ShreyasZare** [reply]
> 
> Also, can you tell me how to make ICANN root DNS to work? Do I need to choose none in quick select option, or just left it empty everywhere? And how much root DNS servers it's using, where I can find it on my VPS?

If you don't specify any forwarders then the DNS server does recursive resolution using root servers. The Quick Select "None" option just clears the text box for you.

**Comment by LexterS999** [comment]
And please think about compression exporting data, because right now it's huge. Need to download 20mb of export config. And pi-hole make it's lightweight

**Reply by ShreyasZare** [reply]
You can exclude logs and block list cache files to reduce the backup zip size. Just check the items you require so the backup size is minimal.
